test/controllers/APIControllerSpec.scala

```diff
@@ -0,0 +1,151 @@
+package controllers
+
+import akka.stream.Materializer
+import akka.util.ByteString
+import org.scalatestplus.play._
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.Play.materializer
+import play.api.test.Helpers._
+import play.api.test.{FakeHeaders, FakeRequest, Injecting}
+import play.api.mvc._
+import java.time.LocalDate
+import java.util.UUID
+import scala.concurrent.Future
+import scala.language.postfixOps
+import play.api.db.Databases
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.streams.Accumulator
+
+class APIControllerSpec
+    extends PlaySpec
+    with Results
+    with GuiceOneAppPerTest
+    with Injecting {
+
+  def mockApp = new GuiceApplicationBuilder().build()
+  val mtrlzr = mockApp.injector.instanceOf[Materializer]
+
+  val database = Databases(
+    driver = "org.h2.Driver",
+    url = "jdbc:h2:mem:play"
+  )
+
+  "postProduct method" should {
+    "return successful when adding a product with all needed values" in {
+      val controller = inject[APIController]
+      val result: Accumulator[ByteString, Result] = controller
+        .postProduct()
+        .apply(
+          FakeRequest(
+            POST,
+            "/pass-bakery/product"
+          ).withBody("""{"name": "crepe", "quantity": 5, "price": 4.99}""")
+            .withHeaders(("Content-Type:", "application/json"))
+        )
+      val runResult: Future[Result] = result.run()(mtrlzr)
+
+      status(runResult) mustBe 200
+    }
+    "return unsuccessful when trying to add a product with no values" in {
+      val controller = inject[APIController]
+      val result: Accumulator[ByteString, Result] = controller
+        .postProduct()
+        .apply(
+          FakeRequest(
+            POST,
+            "/pass-bakery/product"
+          ).withBody("""""")
+            .withHeaders(("Content-Type:", "application/json"))
+        )
+      val runResult: Future[Result] = result.run()(mtrlzr)
+      status(runResult) mustBe 400
+    }
+  }
+  "Edit product method" should {
+
+    "return successful when passing in the ID and some values" in {
+      val controller = inject[APIController]
+      val uuid: UUID = UUID.fromString("a62bf2f7-732d-47a0-b791-3140784784b0")
+
+      val result: Accumulator[ByteString, Result] = controller
+        .editByID(uuid)
+        .apply(
+          FakeRequest(
+            PUT,
+            "/pass-bakery/product"
+          ).withBody("""{"name": "crepe", "quantity": 5, "price": 4.99}""")
+            .withHeaders(("Content-Type:", "application/json"))
+        )
+      val runResult: Future[Result] = result.run()(mtrlzr)
+
+      status(runResult) mustBe 200
+    }
+    "return unsuccessful when passing in an invalid ID" in {
+
+      val controller = inject[APIController]
+      val uuid: UUID = UUID.fromString("a62bf2f7-732d-47a0-b791-3140784704b0")
```

Comment:
it is unclear this is an invalid ID. for me to be convinced, generate one randomly in the test